### PR TITLE
V2 -- Updated Dependencies, Webpack, Hot reloader

### PR DESCRIPTION
**Pull request by stylesuxx**
I have made the needed changes for the v4 gen to work with this PR too, so as soon as this gets pulled, i will open a PR in the other repo.

### CODE REVIEW

**Comment by sthzg** [comment]
Hi @weblogixx @stylesuxx,

could you coordinate these updates between you two? According to the last comment https://github.com/react-webpack-generators/generator-react-webpack/issues/302#issuecomment-261860519 @weblogixx is planning to do the updates. But that might have overlapped with this PR (on a side-note: I need to check the Github email settings, as I don't get emails for new PRs).

@stylesuxx thanks for the PR. I haven't had time to check in detail, but I haven't seen a few of the integrations I had to do to make it work (changes to babelrc, wrapping `Main` in the `<AppContainer />` component, manually triggering the `module.hot.accept()` API). The last time I've set it up was about three weeks ago, do you know if this is obsolete by now? (I still see it in their `next-docs` branch https://github.com/gaearon/react-hot-loader/tree/next-docs/docs, but code might progress more quickly than docs 😄 ).

As said above, as @weblogixx mentioned to update the projects in the linked comment I think its best if you two coordinated it between each other so that there is no double-work on that.

**Comment by stylesuxx** [comment]
OK, I have not seen this comment yet. But it works for me - If @weblogixx has something tested, then I am happy.

Point is, my redux gen is using the V4 of the react gen which uses the latest V2.x template so problems down here are bubbling up on my side and I thought I'll just make the first step to get things rolling ;-)

@sthzg pretty sure I did test hot reloading, but I guess I should double check because I did not make this changes.

Let me know how I can help.

**Comment by weblogixx** [comment]
Hi @stylesuxx,

unfortunately, I do not have the time for an emergency fix today :(. @sthzg: Could you please have a look at it? From the first (5 minute) view, it looks good and may get published.

Thanks and sorry :(

File: src/client.js [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+ReactDOM.render(
+  <AppContainer>
+    <App />
+  </AppContainer>,
+  document.getElementById('app')
+);
+
+if (module.hot) {
+  module.hot.accept('./components/App', () => {
+    const NextApp = require('./components/App').default;
```
Comment (sthzg) [inline src/client.js]: Looking at Travis, this little bugger needs the `// eslint-diasable-line` (to mute the global require() error)

File: src/client.js [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+    ReactDOM.render(
+      <AppContainer>
+        <NextApp/>
```
Comment (sthzg) [inline src/client.js]: And here the airbnb rules mandate the space before closing the tag (=> `<NextApp />`)

**Pull request merged**